tests/lib/rules/static-property-placement.js

```diff
@@ -19,14 +19,16 @@ const PROPERTY_ASSIGNMENT = 'property assignment';
 const RuleTester = require('eslint').RuleTester;
 const rule = require('../../../lib/rules/static-property-placement');
 
-const parsers = require('../../helpers/parsers-old');
+const parsers = require('../../helpers/parsers');
 
 const ruleTesterConfig = {
-  parserOptions: {
+  languageOptions: {
```

Comment:
this seems like it will break on older eslint versions (and eslintrc config), which we still need to retain support for

Reply:
This change was made only in tests and is necessary after upgrading to eslint v9. I don't know if we should keep separate tests for older eslint versions. Maybe we should somehow duplicate all tests to test rules separately with eslint v9 and with older versions.

Reply:
The alternative approach is to use eslint v8 as a dev dependency and test rules with eslint v8 as it was done before, and add v9 to peerDependencies but don't test rules with the latest eslint. I saw this approach in the [eslint-plugin-perfectionist](https://github.com/azat-io/eslint-plugin-perfectionist/tree/main) repo, where eslint v9 support was added but tests are still written in v8. This plugin dropped support for eslint v7 and older as the new API eslint API was introduced in v8.